app/models/operatingsystems/nxos.rb

```diff
@@ -0,0 +1,63 @@
+class NXOS < Operatingsystem
+  # We don't fetch any files here.
+  PXEFILES = {}
+
+  # Simple output of the media url
+  def mediumpath(host)
+    medium_uri(host).to_s
+  end
+
+  def class
+    Operatingsystem
+  end
+
+  # The variant to use when communicating with the proxy. We also need a new type here.
+  def pxe_variant
+    "poap"
+  end
+
+  # The kind of PXE configuration template used. POAP is used for Cisco POAP scripts.
+  def template_kind
+    "POAP"
+  end
+
+  def pxedir
+    "boot/$arch/images"
+  end
+
+  def url_for_boot(file)
+    pxedir + "/" + PXEFILES[file]
+  end
+
+  # where to create the boot file on the TFTP server
+  def boot_filename(host = nil)
+    "poap.cfg/"+host.mac.gsub(/:/,"").upcase
+  end
+
+  def kernel(arch)
```

Comment:
Raise a `Foreman::Exception` here and in initrd perhaps if they are called, rather than returning none.